### PR TITLE
Optimize varint encoding

### PR DESCRIPTION
**Pull request by lpgauth**
As mentioned previously, a lot of time is spent encoding varints when encoding a produce request. Returning `iodata()` instead of a `binary()` saves ~20% and pattern matching seems to save ~5%.

Ideally varint would be a BIF/NIF, but for now this is the best we can do.

### CODE REVIEW

File: src/kpro_varint.erl [hunk 1]
```diff
@@ -38,13 +37,10 @@ enc_zigzag(Int) ->
   (Int bsl 1) bxor (Int bsr ?MAX_BITS).
 
 -spec enc_varint(non_neg_integer()) -> iodata().
+enc_varint(I) when I =< 127 ->
+  [I band 127];
```
Comment (zmstone) [inline src/kpro_varint.erl]:
```suggestion
  [I];
```

File: src/kpro_varint.erl [hunk 2]
```diff
@@ -26,10 +26,9 @@ decode(Bin) ->
...
 %% @doc Encode varint.
--spec encode(kpro:int64()) -> binary().
+-spec encode(kpro:int64()) -> iodata().
 encode(Int) ->
-  Bytes = enc_varint(enc_zigzag(Int)),
-  iolist_to_binary(Bytes).
+  enc_varint(enc_zigzag(Int)).
 
 -spec enc_zigzag(integer()) -> non_neg_integer().
 enc_zigzag(Int) ->
```
Comment (zmstone) [inline src/kpro_varint.erl:35]: add a `-compile({inline, [enc_zigzag/1]}).` to improve a bit more.

**Comment by zmstone** [comment]
There is a `Varint32` in snappy (the dependency of this lib), I'll see if I can add a `Varint64` and expose it as nif.

File: src/kpro_varint.erl (hunk 1 above)
Comment (lpgauth) [reply src/kpro_varint.erl]: Oops, you're right.

**Comment by lpgauth** [comment]
Can you tag a new version? Thanks

**Comment by zmstone** [comment]
2.2.7 is tagged and published to hex.

**Pull request merged**